### PR TITLE
Add format_key and encoding_key to batch

### PR DESCRIPTION
**Pull request by georgiana-b**
Allow setting format and encoding for each source in sources by including  optional `format` and `encoding` columns.

### CODE REVIEW

File: examples/batch/example.csv [hunk 1]
```diff
@@ -1,4 +1,5 @@
-data,schema
-https://raw.githubusercontent.com/okfn/goodtables/master/examples/contacts/people.csv,https://raw.githubusercontent.com/okfn/goodtables/master/examples/contacts/schema_valid.json
-https://raw.githubusercontent.com/okfn/goodtables/master/examples/required_false.csv,https://raw.githubusercontent.com/okfn/goodtables/master/examples/test_schema.json
-https://raw.githubusercontent.com/okfn/goodtables/master/examples/row_limit_structure.csv,
+data,schema,format
+https://raw.githubusercontent.com/okfn/goodtables/master/examples/contacts/people.csv,https://raw.githubusercontent.com/okfn/goodtables/master/examples/contacts/schema_valid.json,csv
+https://raw.githubusercontent.com/okfn/goodtables/master/examples/required_false.csv,https://raw.githubusercontent.com/okfn/goodtables/master/examples/test_schema.json,csv
+https://raw.githubusercontent.com/okfn/goodtables/master/examples/row_limit_structure.csv,,csv
+https://raw.githubusercontent.com/okfn/goodtables/feature/format_and_encoding_key_for_batch/examples/april-to-may-12th-2010.xls,,excel
```
Comment (georgiana-b) [inline examples/batch/example.csv]: Please update this line at merge.

Comment (pwalsh) [inline examples/batch/example.csv]: No, please fix the PR so it can be merged :)

File: goodtables/pipeline/batch.py [hunk 2]
```diff
@@ -116,9 +122,13 @@ def pipeline_factory(self, data, schema):
...
             options['options']['schema'] = {}
-
+            
         options['options']['schema']['schema'] = schema
-
+            
+        if encoding: options['encoding'] = encoding
+        if format:   options['format'] = format
+        
+        print(options)
```
Comment (pwalsh) [inline goodtables/pipeline/batch.py]: Don't leave print functions from debugging in code

**Comment by pwalsh** [comment]
Fix the small comments, then ready to merge

**Comment by pwalsh** [comment]
Also, these changes need to be added to the datapackage handler, here: https://github.com/frictionlessdata/goodtables/blob/master/goodtables/pipeline/batch.py#L105

**Pull request merged**